src/platform/unittest/TlsTest.cpp

```diff
@@ -165,6 +206,21 @@ struct TlsTest : public ::testing::TestWithParam<bool>
         SelfSignedCertParamsFlags = SelfSignedCertParams->Flags;
         ClientCertParams = (QUIC_CREDENTIAL_CONFIG*)CxPlatGetSelfSignedCert(CXPLAT_SELF_SIGN_CERT_USER, TRUE);
         ASSERT_NE(nullptr, ClientCertParams);
+
+        CaSelfSignedCertParams = (QUIC_CREDENTIAL_CONFIG*)CxPlatGetSelfSignedCert(CXPLAT_SELF_SIGN_CA_CERT_USER, FALSE);
+        ASSERT_NE(nullptr, CaSelfSignedCertParams);
+        CaSelfSignedCertParamsFlags = CaSelfSignedCertParams->Flags;
+        CaClientCertParams = (QUIC_CREDENTIAL_CONFIG*)CxPlatGetSelfSignedCert(CXPLAT_SELF_SIGN_CA_CERT_USER, TRUE);
+        ASSERT_NE(nullptr, ClientCertParams);
+
+        // We need to switch the ca roots around since the server must
```

Comment:
Why the swap instead of just assigning the correct certificate to the right *certparams variable?

Reply:
It is tricky for two reasons, the certparams are returned by the CxPlatGetSelfSignedCert, so in order to set them directly the ca certificate file parameter would have to be passed as an argument to CxPlatGetSelfSignedCert. However, the ca certificate file is also created by the CxPlatGetSelfSignedCert so we do not even know what to pass in until at least the "other" version returns. The only solution I see, which I've also considered for other reasons, would be to have a new function that creates both the client and server certificates in one go. That way the ca certificates can be assigned correctly from the start. Should I rework the code with this change?

Reply:
Hmm, maybe the CA certificate creation should be split out into its own function which returns the file path to the CA certificate, and then it can be passed as a parameter to `CxPlatGetSelfSignedCert` to sign the client/server certificate?  I think that would be the most flexible design. Does that sound too complicated?

Reply:
Sounds like a good plan, I'll give it a try and see what it looks like.

Reply:
I pushed a commit with this change, but it introduces quite a bit of code just to avoid the swap...